sqlmesh/core/snapshot/definition.py

```diff
@@ -173,6 +178,13 @@ def is_new_version(self) -> bool:
         """Returns whether or not this version is new and requires a backfill."""
         return self.fingerprint.to_version() == self.version
 
+    @property
+    def indirect_version(self) -> SnapshotIndirectVersion:
```

Comment:
How does adding this property/class relate to the other changes? Or is something you noticed while working on this?

Reply:
By adding this we no longer need to map indirect versions during the migration. The purpose of these is twofold: 
1. It simplifies the migration logic drastically.
2. We no longer need to keep all snapshots in memory in the deserialized form. The mapping logic used to rely on this assumption heavily.